pkg/controller/registry/registry_controller.go

```diff
@@ -242,63 +296,61 @@ func (r *ReconcileRegistry) ensureRegistry(ctx context.Context, harborClient *h.
 
 	// use id from harbor instance
 	if originalRegistry.Spec.ID != heldRegistry.ID {
-		patch := client.MergeFrom(originalRegistry.DeepCopy())
 		originalRegistry.Spec.ID = heldRegistry.ID
+
+		patch := client.MergeFrom(originalRegistry.DeepCopy())
 		if err := r.client.Patch(ctx, originalRegistry, patch); err != nil {
 			return err
 		}
 	}
 
 	if newReg.Credential == nil {
-		newReg.Credential = &h.Credential{}
+		newReg.Credential = &modelv1.RegistryCredential{}
 	}
 	// Compare the registries and update accordingly
 	if !reflect.DeepEqual(heldRegistry, newReg) {
-		return r.updateRegistry(harborClient, newReg)
+		return r.updateRegistry(ctx, harborClient, newReg)
 	}
-	return nil
 
+	return nil
 }
 
-// updateRegistry triggers the update of a registry
-func (r *ReconcileRegistry) updateRegistry(harborClient *h.Client, reg h.Registry) error {
-	return harborClient.Registries().UpdateRegistryByID(reg)
+// updateRegistry triggers the update of a registry.
+func (r *ReconcileRegistry) updateRegistry(ctx context.Context, harborClient *h.RESTClient,
+	reg *modelv1.Registry) error {
+	return harborClient.UpdateRegistry(ctx, reg)
 }
 
-// buildRegistryFromSpec constructs and returns a Harbor registry object from the CR object's spec
-func (r *ReconcileRegistry) buildRegistryFromSpec(originalRegistry *registriesv1alpha1.Registry) (h.Registry, error) {
+// buildRegistryFromSpec constructs and returns a Harbor registry object from the CR object's spec.
+func (r *ReconcileRegistry) buildRegistryFromSpec(originalRegistry *registriesv1alpha1.Registry) (*modelv1.Registry,
+	error) {
 	parsedURL, err := parseURL(originalRegistry.Spec.URL)
-
 	if err != nil {
-		return h.Registry{}, err
+		return &modelv1.Registry{}, err
```

Comment:
Should be OK to just return `nil` in this case.

```suggestion
		return nil, err
```